### DISCUSSION TITLE
[Feature] Use Bun

### DISCUSSION THREAD

**Discussion by gaurishhs** [immich-app/immich#5289] [Feature Request]
### The feature

Hi! [Bun](https://bun.sh) is a faster js runtime. Migrating Immich to Bun can reduce the memory and CPU usage by a lot. Bun does support NestJS but the only thing that could stop this migration is no TypeORM support but that is being worked on https://github.com/oven-sh/bun/issues/4136 but Bun does support Prisma, Sequelize, Drizzle and many others even MikroORM with some workaround

Overall, Migrating to Bun is definitely going to be beneficial for Immich and greatly improve the user experience. 

Thanks!

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by mertalev** [comment]
We've looked into Bun, but we'd like to wait for it to mature before switching to it. There were many issues that came up that individually had workarounds, but collectively made it clear that Bun wasn't ready yet.

**Reply by gaurishhs** [reply]
What issues did you face with Bun?

**Reply by aymanbazyan** [reply]
@mertalev 
Hello, I think v1.3.9 is good and fast enough now, maybe you should look again?
